app/controllers/employment/CheckEmploymentDetailsController.scala

```diff
@@ -18,43 +18,62 @@ package controllers.employment
 
 import audit.{AuditService, ViewEmploymentDetailsAudit}
 import config.AppConfig
-import controllers.predicates.AuthorisedAction
+import controllers.predicates.{AuthorisedAction, InYearAction}
 import models.employment.{AllEmploymentData, EmploymentSource}
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 import utils.SessionHelper
 import views.html.employment.CheckEmploymentDetailsView
+
 import javax.inject.Inject
 import services.IncomeTaxUserDataService
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
 class CheckEmploymentDetailsController @Inject()(implicit val cc: MessagesControllerComponents,
                                                  authAction: AuthorisedAction,
+                                                 inYearAction: InYearAction,
                                                  employmentDetailsView: CheckEmploymentDetailsView,
-                                                 implicit val appConfig: AppConfig,
+                                                 appConfig: AppConfig,
                                                  incomeTaxUserDataService: IncomeTaxUserDataService,
                                                  auditService: AuditService,
-                                                 implicit val ec: ExecutionContext) extends FrontendController(cc) with I18nSupport with SessionHelper {
+                                                 ec: ExecutionContext) extends FrontendController(cc) with I18nSupport with SessionHelper {
 
 
   def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
 
     def result(allEmploymentData: AllEmploymentData): Result = {
 
-      val source: Option[EmploymentSource] = allEmploymentData.hmrcEmploymentData.find(source => source.employmentId.equals(employmentId))
+      val customerData: (Option[EmploymentSource], Boolean) = getCustomerData(allEmploymentData, employmentId, taxYear)
```

Comment:
I'd change customerData to be only `allEmploymentData.customerEmploymentData.find(source => source.employmentId.equals(employmentId))`

and then when you do the if statements on line 50 use `if(customerData.isDefined && !inYearAction.inYear(taxYear))` instead.

could define a val like isUsingCustomerData that equals `customerData.isDefined && !inYearAction.inYear(taxYear)`

Reply:
agreed -changed